server/models/db.js

```diff
@@ -0,0 +1,16 @@
+const mongoose = require("mongoose");
+
+const connectDB = async () => {
+  try {
+    // mongoose setup
+    mongoose.connect(process.env.DBSTRING);
+    const db = mongoose.connection;
+    db.on("error", (error) => console.log(error));
+    db.once("open", () => console.log("Connected to db :)"));
```

Comment:
The `on` and `once` event handlers aren't necessary here since the `connectDB` function is responsible for the initial connection to the MongoDB server. The `connect` promise will reject if there's an error (which you're already handling with `try/catch`). Changing to `await` will resolve the promise (so no need for waiting for the `open` event).

```suggestion
    const conn = await mongoose.connect(process.env.DBSTRING);   
    console.log(`MongoDB Connected: ${conn.connection.host}`)
```